pkg/auth/handler/webapp/wechat_auth.go

```diff
@@ -143,6 +129,33 @@ func (h *WechatAuthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		h.Renderer.RenderHTML(w, r, TemplateWebWechatAuthHandlerHTML, data)
 		return nil
 	})
+
+	// Send WebsocketMessageKindWeChatLoginStart event
+	ctrl.PostAction("", func() error {
+		session, err := ctrl.InteractionSession()
+		if err != nil {
+			return err
+		}
+
+		msg := &WebsocketMessage{
+			Kind: WebsocketMessageKindWeChatLoginStart,
+			Data: WebsocketMessageWeChatLoginStartData{
+				State: session.ID,
+			},
+		}
+
+		err = h.Publisher.Publish(session, msg)
+		if err != nil {
+			return err
+		}
+
+		// stay in the same page, no navigation action
+		result := &webapp.Result{
+			NavigationAction: "no_action",
```

Comment:
I think we should also set webapp.Result.RedirectURI to current URI, in case turbolinks is not being used. We can test this by disabling JavaScript in the browser.

Reply:
The commit has been removed and handled in #857